Index: nodogsplash-1.0.2/src/fw_iptables.c
===================================================================
--- a/src/fw_iptables.c
+++ b/src/fw_iptables.c
@@ -598,7 +598,7 @@ iptables_fw_init(void)
 		rc |= _iptables_append_ruleset("filter", "preauthenticated-users", CHAIN_TO_INTERNET);
 	}
 	/* CHAIN_TO_INTERNET, all other packets REJECT */
-	rc |= iptables_do_command("-t filter -A " CHAIN_TO_INTERNET " -j REJECT --reject-with icmp-port-unreachable");
+	//rc |= iptables_do_command("-t filter -A " CHAIN_TO_INTERNET " -j REJECT --reject-with icmp-port-unreachable");
 
 	/*
 	 * End of filter table chains and rules
